membership-common/src/test/scala/com/gu/memsub/subsv2/reads/SubReadsTest.scala

```diff
@@ -2,47 +2,40 @@ package com.gu.memsub.subsv2.reads
 
 import com.gu.i18n.Currency._
 import com.gu.lib.DateDSL._
-import com.gu.memsub.Benefit.Weekly
-import com.gu.memsub.BillingPeriod.Quarter
-import com.gu.memsub.Product.WeeklyDomestic
 import com.gu.memsub.Subscription.{ProductRatePlanChargeId, ProductRatePlanId, RatePlanId, SubscriptionRatePlanChargeId}
-import com.gu.memsub._
 import com.gu.memsub.subsv2.ReaderType.Patron
 import com.gu.memsub.subsv2._
 import com.gu.memsub.subsv2.reads.SubJsonReads._
-import org.joda.time.LocalDate
+import com.gu.memsub.{subsv2, _}
 import org.specs2.mutable.Specification
-import utils.Resource
 import scalaz.NonEmptyList
+import utils.Resource
 
 class SubReadsTest extends Specification {
 
   "Subscription JSON reads" should {
 
     "Discard discount rate plans when reading JSON" in {
```

Comment:
this only discards them because the Pricing Summary can't parse discounts so the whole rateplan is discarded.
If it's a discount amount instead, it wil be retained.  This test will have to change, probably in a later PR.
https://github.com/guardian/members-data-api/blob/729767723fe998ad888fc8858f83e9619b0e2bad/membership-common/src/test/resources/rest/plans/Promo.json#L64